src/pkg/tag/manager.go

```diff
@@ -33,14 +34,47 @@ type Manager interface {
 	Get(ctx context.Context, id int64) (tag *tag.Tag, err error)
 	// Create the tag and returns the ID
 	Create(ctx context.Context, tag *tag.Tag) (id int64, err error)
-	// Update the tag
-	Update(ctx context.Context, tag *tag.Tag) (err error)
+	// Update the tag. Only the properties specified by "props" will be updated if it is set
+	Update(ctx context.Context, tag *tag.Tag, props ...string) (err error)
 	// Delete the tag specified by ID
 	Delete(ctx context.Context, id int64) (err error)
 }
 
 // NewManager creates an instance of the default tag manager
 func NewManager() Manager {
-	// TODO implement
-	return nil
+	return &manager{
+		dao: dao.New(),
+	}
+}
+
+type manager struct {
+	dao dao.DAO
+}
+
+func (m *manager) List(ctx context.Context, query *q.Query) (int64, []*tag.Tag, error) {
+	total, err := m.dao.Count(ctx, query)
+	if err != nil {
+		return 0, nil, err
+	}
+	tags, err := m.dao.List(ctx, query)
+	if err != nil {
+		return 0, nil, err
+	}
+	return total, tags, nil
+}
+
+func (m *manager) Get(ctx context.Context, id int64) (*tag.Tag, error) {
+	return m.dao.Get(ctx, id)
+}
+
+func (m *manager) Create(ctx context.Context, tag *tag.Tag) (int64, error) {
+	return m.dao.Create(ctx, tag)
+}
+
+func (m *manager) Update(ctx context.Context, tag *tag.Tag, props ...string) error {
```

Comment:
Doesn't make sense to ask caller of manager to specify the name of the columns in DB.

Reply:
It isn't the DB column, it's the name of model property, e.g. `PushTime` rather than `push_time`

Reply:
But still this is tied to the implementation of orm.  I wish to suggest we restrict the props to avoid error.  Otherwise it even supports to move one tag to another artifact which will never happen.